libraries/bsg_manycore_epa.h

```diff
@@ -51,6 +54,33 @@ extern "C" {
 #define EPA_FROM_BASE_AND_OFFSET(base, offset)  \
         (((base)+(offset)))
 
+        /**
+         * Checks alignment of an epa based on data size in bytes.
+         * @param[in] epa  epa address
+         * @param[in] sz   data size in bytes.
+         * @return         HB_MC_SUCCESS if npa is aligned and HB_MC_INVALID otherwise.
+         */
+        static inline int hb_mc_epa_check_alignment(const hb_mc_epa_t *epa, size_t sz)
+        {
+                switch (sz) {
```

Comment:
A shorter way to implement this could be to check that `sz` is 1, 2, or 4, and then check that `epa & (sz - 1) == 0`, though the way it is now may be more readable :)

Reply:
Yes, but then you'd need to validate sz.